goal: optimize roc_auc

note: batch_size paramter is later removed in utils for method run_5fold

In [18]:
from utils import max_features, maxlen, embed_size_fastText, embed_size_glove, embed_size_glove_twitter
from utils import earlystop, batch_size, epochs, run_5fold
from model import get_CNN_model
import pickle
import tensorflow as tf

In [2]:
fastText = pickle.load(open("input/fastText.300d.pickle", "rb"))
glove = pickle.load(open("input/glove.300d.pickle", "rb"))
glove_twitter = pickle.load(open("input/glove.twitter.200d.pickle", "rb"))

In [3]:
x_train = pickle.load(open("input/x_train_small.pickle", "rb"))
y_train = pickle.load(open("input/y_train_small.pickle", "rb"))

## check different embedding

In [4]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "dropout": 0.2,
    "num_filter": 300,
    "kernel_size": 3,
    "reg": 0.0
}

In [5]:
embeddings = [(fastText, embed_size_fastText, "fastText"), (glove, embed_size_glove, "glove"), (glove_twitter, embed_size_glove_twitter, "glove_twitter")]

In [7]:
for embed in embeddings:
    print ("\n********************{}********************".format(embed[2]))
    params["embedding_matrix"] = embed[0]
    params["embed_size"] = embed[1]
    run_5fold(x_train, y_train, batch_size, get_CNN_model, **params)

********************fastText********************
Instructions for updating:
Please switch to tf.metrics.auc. Note that the order of the labels and predictions arguments has been switched.
Epoch 1/20
 - 52s - loss: 0.1579 - acc: 0.9535 - auc_roc: 0.6579
Epoch 2/20
 - 4s - loss: 0.1012 - acc: 0.9688 - auc_roc: 0.7946
Epoch 3/20
 - 4s - loss: 0.0721 - acc: 0.9773 - auc_roc: 0.8554
Epoch 4/20
 - 4s - loss: 0.0514 - acc: 0.9822 - auc_roc: 0.8943
Epoch 5/20
 - 4s - loss: 0.0380 - acc: 0.9865 - auc_roc: 0.9209
Epoch 6/20
 - 4s - loss: 0.0283 - acc: 0.9900 - auc_roc: 0.9395
Epoch 7/20
 - 4s - loss: 0.0211 - acc: 0.9929 - auc_roc: 0.9524
Epoch 8/20
 - 4s - loss: 0.0159 - acc: 0.9951 - auc_roc: 0.9616
Epoch 9/20
 - 4s - loss: 0.0118 - acc: 0.9968 - auc_roc: 0.9684
Epoch 10/20
 - 4s - loss: 0.0089 - acc: 0.9980 - auc_roc: 0.9734
Epoch 11/20
 - 4s - loss: 0.0066 - acc: 0.9988 - auc_roc: 0.9773
Epoch 12/20
 - 4s - loss: 0.0051 - acc: 0.9993 - auc_roc: 0.9804
Epoch 13/20
 - 4s - loss: 0.0040 - acc: 

Epoch 17/20
 - 4s - loss: 0.0017 - acc: 0.9998 - auc_roc: 0.9924
Epoch 18/20
 - 4s - loss: 0.0016 - acc: 0.9999 - auc_roc: 0.9931
Epoch 19/20
 - 4s - loss: 0.0014 - acc: 0.9999 - auc_roc: 0.9937
Epoch 20/20
 - 4s - loss: 0.0012 - acc: 0.9999 - auc_roc: 0.9942
acc: 97.8473386340%
auc_roc: 99.3187489568%
acc: 97.7992929902% (+/- 0.0551891010%)
roc_auc: 99.2331722337% (+/- 0.0605756470%)
********************glove_twitter********************
Epoch 1/20
 - 4s - loss: 0.1391 - acc: 0.9609 - auc_roc: 0.7104
Epoch 2/20
 - 3s - loss: 0.0894 - acc: 0.9722 - auc_roc: 0.8398
Epoch 3/20
 - 3s - loss: 0.0645 - acc: 0.9788 - auc_roc: 0.8906
Epoch 4/20
 - 3s - loss: 0.0488 - acc: 0.9833 - auc_roc: 0.9200
Epoch 5/20
 - 3s - loss: 0.0361 - acc: 0.9876 - auc_roc: 0.9396
Epoch 6/20
 - 3s - loss: 0.0275 - acc: 0.9909 - auc_roc: 0.9531
Epoch 7/20
 - 3s - loss: 0.0213 - acc: 0.9934 - auc_roc: 0.9628
Epoch 8/20
 - 3s - loss: 0.0156 - acc: 0.9955 - auc_roc: 0.9697
Epoch 9/20
 - 3s - loss: 0.0122 - acc: 0.9967 

##### result
- fastText: 

acc: 97.8086930503% (+/- 0.0257543607%)

roc_auc: 99.1276712434% (+/- 0.0649324891%)

- glove: 

acc: 97.7992929902% (+/- 0.0551891010%)

roc_auc: 99.2331722337% (+/- 0.0605756470%)

- glove.twitter: 

acc: 97.7909369218% (+/- 0.0314384256%)

roc_auc: 99.1373431382% (+/- 0.0269865331%)

=============================================

- mini conclusion:

multi channel: fastText, glove

# single model for fastText

## check num_filter
(suggest values: 100~600: test 100, 300, 500)

In [8]:
filters = [100, 500]  # for the result of 300, refer to the previous log

In [9]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "dropout": 0.2,
    "kernel_size": 3,
    "reg": 0.0,
    "embedding_matrix": fastText,
    "embed_size": 300
}

In [12]:
for f in filters:
    print ("\n********************f = {}********************".format(f))
    params["num_filter"] = f
    run_5fold(x_train, y_train, batch_size, get_CNN_model, **params)


********************f = 100********************
Epoch 1/20
 - 3s - loss: 0.1639 - acc: 0.9512 - auc_roc: 0.6508
Epoch 2/20
 - 3s - loss: 0.1209 - acc: 0.9640 - auc_roc: 0.7725
Epoch 3/20
 - 3s - loss: 0.0953 - acc: 0.9694 - auc_roc: 0.8224
Epoch 4/20
 - 3s - loss: 0.0675 - acc: 0.9777 - auc_roc: 0.8627
Epoch 5/20
 - 3s - loss: 0.0496 - acc: 0.9826 - auc_roc: 0.8952
Epoch 6/20
 - 3s - loss: 0.0385 - acc: 0.9865 - auc_roc: 0.9184
Epoch 7/20
 - 3s - loss: 0.0303 - acc: 0.9893 - auc_roc: 0.9350
Epoch 8/20
 - 3s - loss: 0.0233 - acc: 0.9922 - auc_roc: 0.9473
Epoch 9/20
 - 3s - loss: 0.0186 - acc: 0.9942 - auc_roc: 0.9565
Epoch 10/20
 - 3s - loss: 0.0145 - acc: 0.9960 - auc_roc: 0.9634
Epoch 11/20
 - 3s - loss: 0.0116 - acc: 0.9970 - auc_roc: 0.9689
Epoch 12/20
 - 3s - loss: 0.0094 - acc: 0.9980 - auc_roc: 0.9731
Epoch 13/20
 - 3s - loss: 0.0077 - acc: 0.9983 - auc_roc: 0.9766
Epoch 14/20
 - 3s - loss: 0.0059 - acc: 0.9992 - auc_roc: 0.9794
Epoch 15/20
 - 3s - loss: 0.0049 - acc: 0.9994 - a

###### result

- f = 100

acc: 97.7773590230% (+/- 0.0493671869%)

roc_auc: 98.7483467499% (+/- 0.1451896848%)

- f = 300

acc: 97.8086930503% (+/- 0.0257543607%)

roc_auc: 99.1276712434% (+/- 0.0649324891%)

- f = 500

acc: 97.7992929424% (+/- 0.0484303212%)

roc_auc: 99.1831751375% (+/- 0.0707826849%)

## check kernel_size
(suggest value: [3, 5, 7, 9])

In [15]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "dropout": 0.2,
    "num_filter": 500,
    "reg": 0.0,
    "embedding_matrix": fastText,
    "embed_size": 300
}

In [16]:
kernels = [5, 7, 9]  # for 3, see result above

In [20]:
for k in kernels:
    print ("\n********************k = {}********************".format(k))
    params["kernel_size"] = k
    run_5fold(x_train, y_train, batch_size, get_CNN_model, **params)    


********************k = 5********************
Epoch 1/20
 - 14s - loss: 0.1493 - acc: 0.9578 - auc_roc: 0.7000
Epoch 2/20
 - 13s - loss: 0.0941 - acc: 0.9703 - auc_roc: 0.8196
Epoch 3/20
 - 13s - loss: 0.0605 - acc: 0.9802 - auc_roc: 0.8784
Epoch 4/20
 - 13s - loss: 0.0406 - acc: 0.9859 - auc_roc: 0.9147
Epoch 5/20
 - 13s - loss: 0.0282 - acc: 0.9903 - auc_roc: 0.9383
Epoch 6/20
 - 13s - loss: 0.0196 - acc: 0.9937 - auc_roc: 0.9534
Epoch 7/20
 - 13s - loss: 0.0134 - acc: 0.9962 - auc_roc: 0.9635
Epoch 8/20
 - 13s - loss: 0.0092 - acc: 0.9978 - auc_roc: 0.9708
Epoch 9/20
 - 13s - loss: 0.0066 - acc: 0.9989 - auc_roc: 0.9760
Epoch 10/20
 - 13s - loss: 0.0046 - acc: 0.9993 - auc_roc: 0.9799
Epoch 11/20
 - 13s - loss: 0.0038 - acc: 0.9996 - auc_roc: 0.9828
Epoch 12/20
 - 13s - loss: 0.0029 - acc: 0.9997 - auc_roc: 0.9851
Epoch 13/20
 - 13s - loss: 0.0021 - acc: 0.9999 - auc_roc: 0.9870
Epoch 14/20
 - 13s - loss: 0.0020 - acc: 0.9998 - auc_roc: 0.9885
Epoch 15/20
 - 13s - loss: 0.0016 - ac

Epoch 20/20
 - 13s - loss: 0.0010 - acc: 0.9999 - auc_roc: 0.9942
acc: 97.7878033588%
auc_roc: 99.3056515281%
acc: 97.7376688234% (+/- 0.0522443225%)
roc_auc: 99.2505021100% (+/- 0.0420800186%)

********************k = 9********************
Epoch 1/20
 - 15s - loss: 0.1483 - acc: 0.9569 - auc_roc: 0.7114
Epoch 2/20
 - 14s - loss: 0.0912 - acc: 0.9708 - auc_roc: 0.8187
Epoch 3/20
 - 14s - loss: 0.0551 - acc: 0.9816 - auc_roc: 0.8822
Epoch 4/20
 - 14s - loss: 0.0353 - acc: 0.9876 - auc_roc: 0.9191
Epoch 5/20
 - 14s - loss: 0.0238 - acc: 0.9922 - auc_roc: 0.9422
Epoch 6/20
 - 14s - loss: 0.0154 - acc: 0.9955 - auc_roc: 0.9565
Epoch 7/20
 - 14s - loss: 0.0103 - acc: 0.9975 - auc_roc: 0.9662
Epoch 8/20
 - 14s - loss: 0.0070 - acc: 0.9987 - auc_roc: 0.9729
Epoch 9/20
 - 14s - loss: 0.0047 - acc: 0.9994 - auc_roc: 0.9777
Epoch 10/20
 - 14s - loss: 0.0033 - acc: 0.9997 - auc_roc: 0.9812
Epoch 11/20
 - 14s - loss: 0.0029 - acc: 0.9997 - auc_roc: 0.9839
Epoch 12/20
 - 14s - loss: 0.0021 - acc: 0

##### result:

- k = 3

acc: 97.7992929424% (+/- 0.0484303212%)

roc_auc: 99.1831751375% (+/- 0.0707826849%)

- k = 5

acc: 97.7710921528% (+/- 0.0348296133%)

roc_auc: 99.2507335969% (+/- 0.0359207103%)

- k = 7

acc: 97.7376688234% (+/- 0.0522443225%)

roc_auc: 99.2505021100% (+/- 0.0420800186%)

- k = 9

acc: 97.7522914591% (+/- 0.0299820318%)

roc_auc: 99.2583731168% (+/- 0.0241783867%)


## check drop out
(suggest value 0.1~0.5: test 0.1, 0.2, 0.5)

In [21]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "kernel_size": 9,
    "num_filter": 500,
    "reg": 0.0,
    "embedding_matrix": fastText,
    "embed_size": 300
}

In [23]:
dropouts = [0.1, 0.5] # 0.2 see previous result

In [24]:
for d in dropouts:
    print ("\n********************d = {}********************".format(d))
    params["dropout"] = d
    run_5fold(x_train, y_train, batch_size, get_CNN_model, **params)    


********************d = 0.1********************
Epoch 1/20
 - 14s - loss: 0.1444 - acc: 0.9571 - auc_roc: 0.7175
Epoch 2/20
 - 14s - loss: 0.0840 - acc: 0.9733 - auc_roc: 0.8402
Epoch 3/20
 - 14s - loss: 0.0498 - acc: 0.9832 - auc_roc: 0.8982
Epoch 4/20
 - 14s - loss: 0.0298 - acc: 0.9901 - auc_roc: 0.9323
Epoch 5/20
 - 14s - loss: 0.0187 - acc: 0.9945 - auc_roc: 0.9523
Epoch 6/20
 - 14s - loss: 0.0124 - acc: 0.9970 - auc_roc: 0.9646
Epoch 7/20
 - 14s - loss: 0.0080 - acc: 0.9983 - auc_roc: 0.9727
Epoch 8/20
 - 14s - loss: 0.0054 - acc: 0.9994 - auc_roc: 0.9781
Epoch 9/20
 - 14s - loss: 0.0037 - acc: 0.9996 - auc_roc: 0.9821
Epoch 10/20
 - 14s - loss: 0.0029 - acc: 0.9997 - auc_roc: 0.9850
Epoch 11/20
 - 14s - loss: 0.0023 - acc: 0.9998 - auc_roc: 0.9871
Epoch 12/20
 - 14s - loss: 0.0020 - acc: 0.9998 - auc_roc: 0.9888
Epoch 13/20
 - 14s - loss: 0.0016 - acc: 0.9999 - auc_roc: 0.9902
Epoch 14/20
 - 14s - loss: 0.0012 - acc: 0.9999 - auc_roc: 0.9913
Epoch 15/20
 - 14s - loss: 0.0013 - 

Epoch 20/20
 - 14s - loss: 0.0019 - acc: 0.9997 - auc_roc: 0.9905
acc: 97.8912063617%
auc_roc: 98.9490015163%
acc: 97.8421162840% (+/- 0.0423750447%)
roc_auc: 98.9288327762% (+/- 0.0654320279%)


##### result

- d = 0.1

acc: 97.7272242339% (+/- 0.0262575199%)

roc_auc: 99.3181312003% (+/- 0.0253788972%)

- d = 0.2

acc: 97.7522914591% (+/- 0.0299820318%)

roc_auc: 99.2583731168% (+/- 0.0241783867%)

- d = 0.5

acc: 97.8421162840% (+/- 0.0423750447%)

roc_auc: 98.9288327762% (+/- 0.0654320279%)

#### result params for fastText

In [28]:
params_fastText = {
    "maxlen": maxlen,
    "max_features": max_features,
    "kernel_size": 9,
    "num_filter": 500,
    "reg": 0.0,
    "embedding_matrix": fastText,
    "embed_size": 300, 
    "dropout": 0.1
}

# single model for glove

## check num_filter
(suggest values: 100~600: test 100, 300, 500)

In [26]:
filters = [100, 500]  # for the result of 300, refer to the previous log

In [27]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "dropout": 0.2,
    "kernel_size": 3,
    "reg": 0.0,
    "embedding_matrix": glove,
    "embed_size": 300
}

In [29]:
for f in filters:
    print ("\n********************f = {}********************".format(f))
    params["num_filter"] = f
    run_5fold(x_train, y_train, batch_size, get_CNN_model, **params)


********************f = 100********************
Epoch 1/20
 - 3s - loss: 0.1486 - acc: 0.9591 - auc_roc: 0.6659
Epoch 2/20
 - 3s - loss: 0.1049 - acc: 0.9674 - auc_roc: 0.8014
Epoch 3/20
 - 3s - loss: 0.0727 - acc: 0.9770 - auc_roc: 0.8595
Epoch 4/20
 - 3s - loss: 0.0538 - acc: 0.9815 - auc_roc: 0.8985
Epoch 5/20
 - 3s - loss: 0.0408 - acc: 0.9857 - auc_roc: 0.9237
Epoch 6/20
 - 3s - loss: 0.0320 - acc: 0.9888 - auc_roc: 0.9411
Epoch 7/20
 - 3s - loss: 0.0243 - acc: 0.9920 - auc_roc: 0.9533
Epoch 8/20
 - 3s - loss: 0.0193 - acc: 0.9941 - auc_roc: 0.9621
Epoch 9/20
 - 3s - loss: 0.0148 - acc: 0.9961 - auc_roc: 0.9687
Epoch 10/20
 - 3s - loss: 0.0116 - acc: 0.9971 - auc_roc: 0.9737
Epoch 11/20
 - 3s - loss: 0.0093 - acc: 0.9981 - auc_roc: 0.9776
Epoch 12/20
 - 3s - loss: 0.0077 - acc: 0.9985 - auc_roc: 0.9807
Epoch 13/20
 - 3s - loss: 0.0062 - acc: 0.9991 - auc_roc: 0.9831
Epoch 14/20
 - 3s - loss: 0.0051 - acc: 0.9994 - auc_roc: 0.9851
Epoch 15/20
 - 3s - loss: 0.0042 - acc: 0.9995 - a

###### result:

- 100:

acc: 97.7972040030% (+/- 0.0468966292%)

roc_auc: 98.9772399382% (+/- 0.0936338543%)

- 300:

acc: 97.7992929902% (+/- 0.0551891010%)

roc_auc: 99.2331722337% (+/- 0.0605756470%)

- 500:

acc: 97.8013818340% (+/- 0.0860913550%)

roc_auc: 99.2587193698% (+/- 0.0388054175%)

## check kernel size
(suggest value: [3, 5, 7, 9])

In [30]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "dropout": 0.2,
    "num_filter": 500,
    "reg": 0.0,
    "embedding_matrix": glove,
    "embed_size": 300
}

In [31]:
kernels = [5, 7, 9]  # for 3, see result above

In [32]:
for k in kernels:
    print ("\n********************k = {}********************".format(k))
    params["kernel_size"] = k
    run_5fold(x_train, y_train, batch_size, get_CNN_model, **params)    


********************k = 5********************
Epoch 1/20
 - 12s - loss: 0.1511 - acc: 0.9581 - auc_roc: 0.6872
Epoch 2/20
 - 12s - loss: 0.0816 - acc: 0.9742 - auc_roc: 0.8264
Epoch 3/20
 - 12s - loss: 0.0505 - acc: 0.9829 - auc_roc: 0.8895
Epoch 4/20
 - 12s - loss: 0.0326 - acc: 0.9893 - auc_roc: 0.9250
Epoch 5/20
 - 12s - loss: 0.0197 - acc: 0.9941 - auc_roc: 0.9465
Epoch 6/20
 - 12s - loss: 0.0120 - acc: 0.9974 - auc_roc: 0.9601
Epoch 7/20
 - 12s - loss: 0.0080 - acc: 0.9989 - auc_roc: 0.9692
Epoch 8/20
 - 12s - loss: 0.0052 - acc: 0.9994 - auc_roc: 0.9752
Epoch 9/20
 - 12s - loss: 0.0039 - acc: 0.9996 - auc_roc: 0.9796
Epoch 10/20
 - 12s - loss: 0.0031 - acc: 0.9996 - auc_roc: 0.9829
Epoch 11/20
 - 12s - loss: 0.0025 - acc: 0.9998 - auc_roc: 0.9853
Epoch 12/20
 - 12s - loss: 0.0021 - acc: 0.9998 - auc_roc: 0.9872
Epoch 13/20
 - 12s - loss: 0.0017 - acc: 0.9999 - auc_roc: 0.9887
Epoch 14/20
 - 12s - loss: 0.0017 - acc: 0.9998 - auc_roc: 0.9899
Epoch 15/20
 - 12s - loss: 0.0017 - ac

Epoch 20/20
 - 9s - loss: 0.0019 - acc: 0.9999 - auc_roc: 0.9955
acc: 97.7564695187%
auc_roc: 99.4109903406%
acc: 97.7449803026% (+/- 0.0835450620%)
roc_auc: 99.3592604342% (+/- 0.0654025651%)

********************k = 9********************
Epoch 1/20
 - 10s - loss: 0.1468 - acc: 0.9606 - auc_roc: 0.7184
Epoch 2/20
 - 10s - loss: 0.0757 - acc: 0.9757 - auc_roc: 0.8538
Epoch 3/20
 - 10s - loss: 0.0431 - acc: 0.9857 - auc_roc: 0.9083
Epoch 4/20
 - 10s - loss: 0.0229 - acc: 0.9927 - auc_roc: 0.9392
Epoch 5/20
 - 10s - loss: 0.0131 - acc: 0.9968 - auc_roc: 0.9576
Epoch 6/20
 - 10s - loss: 0.0080 - acc: 0.9986 - auc_roc: 0.9686
Epoch 7/20
 - 10s - loss: 0.0056 - acc: 0.9992 - auc_roc: 0.9756
Epoch 8/20
 - 16s - loss: 0.0037 - acc: 0.9995 - auc_roc: 0.9805
Epoch 9/20
 - 16s - loss: 0.0031 - acc: 0.9996 - auc_roc: 0.9837
Epoch 10/20
 - 16s - loss: 0.0024 - acc: 0.9998 - auc_roc: 0.9862
Epoch 11/20
 - 16s - loss: 0.0022 - acc: 0.9998 - auc_roc: 0.9880
Epoch 12/20
 - 16s - loss: 0.0021 - acc: 0.

##### result 

- 3:

acc: 97.8013818340% (+/- 0.0860913550%)

roc_auc: 99.2587193698% (+/- 0.0388054175%)

- 5: 

acc: 97.7585583610% (+/- 0.0420651162%)

roc_auc: 99.3059665452% (+/- 0.0606478754%)

- 7: 

acc: 97.7449803026% (+/- 0.0835450620%)

roc_auc: 99.3592604342% (+/- 0.0654025651%)

- 9:

acc: 97.7084236836% (+/- 0.0564208832%)

roc_auc: 99.3585640975% (+/- 0.0253187877%)

## check dropout

In [33]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "kernel_size": 7,
    "num_filter": 500,
    "reg": 0.0,
    "embedding_matrix": glove,
    "embed_size": 300
}

In [34]:
dropouts = [0.1, 0.5] # 0.2 see previous result

In [35]:
for d in dropouts:
    print ("\n********************d = {}********************".format(d))
    params["dropout"] = d
    run_5fold(x_train, y_train, batch_size, get_CNN_model, **params)    


********************d = 0.1********************
Epoch 1/20
 - 14s - loss: 0.1447 - acc: 0.9619 - auc_roc: 0.6930
Epoch 2/20
 - 14s - loss: 0.0754 - acc: 0.9760 - auc_roc: 0.8414
Epoch 3/20
 - 14s - loss: 0.0436 - acc: 0.9857 - auc_roc: 0.9027
Epoch 4/20
 - 14s - loss: 0.0221 - acc: 0.9937 - auc_roc: 0.9358
Epoch 5/20
 - 14s - loss: 0.0118 - acc: 0.9974 - auc_roc: 0.9556
Epoch 6/20
 - 14s - loss: 0.0067 - acc: 0.9990 - auc_roc: 0.9674
Epoch 7/20
 - 14s - loss: 0.0047 - acc: 0.9995 - auc_roc: 0.9748
Epoch 8/20
 - 14s - loss: 0.0035 - acc: 0.9997 - auc_roc: 0.9798
Epoch 9/20
 - 14s - loss: 0.0023 - acc: 0.9998 - auc_roc: 0.9833
Epoch 10/20
 - 14s - loss: 0.0023 - acc: 0.9998 - auc_roc: 0.9858
Epoch 11/20
 - 14s - loss: 0.0020 - acc: 0.9998 - auc_roc: 0.9877
Epoch 12/20
 - 14s - loss: 0.0016 - acc: 0.9998 - auc_roc: 0.9892
Epoch 13/20
 - 14s - loss: 0.0016 - acc: 0.9998 - auc_roc: 0.9905
Epoch 14/20
 - 14s - loss: 0.0017 - acc: 0.9998 - auc_roc: 0.9914
Epoch 15/20
 - 14s - loss: 0.0016 - 

Epoch 20/20
 - 14s - loss: 0.0020 - acc: 0.9997 - auc_roc: 0.9913
acc: 97.8066041721%
auc_roc: 99.0227038339%
acc: 97.8013819297% (+/- 0.0474747065%)
roc_auc: 99.0310144613% (+/- 0.0366966955%)


##### result
- 0.1:

acc: 97.6656003420% (+/- 0.0393864928%)

roc_auc: 99.3375334019% (+/- 0.0405904593%)

- 0.2:

acc: 97.7449803026% (+/- 0.0835450620%)

roc_auc: 99.3592604342% (+/- 0.0654025651%)


- 0.5:

acc: 97.8013819297% (+/- 0.0474747065%)

roc_auc: 99.0310144613% (+/- 0.0366966955%)

##### params for glove

In [36]:
params = {
    "maxlen": maxlen,
    "max_features": max_features,
    "kernel_size": 7,
    "num_filter": 500,
    "reg": 0.0,
    "embedding_matrix": glove,
    "embed_size": 300, 
    "dropout": 0.2
}